### PR TITLE
[release-1.0] OCPBUGS-22347: address CVE-2023-44487

### PR DESCRIPTION
**Pull request by alebedev87**
Backport of https://github.com/openshift/external-dns-operator/pull/204:
- bump `x/net` `v0.17.0`
- bump sigs.k8s.io/controller-runtime v0.12.3 (TLSOpts for webhook)
- bump operand image
- disable HTTP/2 for kube-rbac-proxy container
- disable HTTP/2 for webhook server

### CODE REVIEW

**Comment by alebedev87** [comment]
/title [WIP] [release-1.0] OCPBUGS-22347: address CVE-2023-44487

Waiting for [the operand PR](https://github.com/openshift/external-dns/pull/56) to be merged.

**Comment by alebedev87** [comment]
/retitle [WIP] [release-1.0] OCPBUGS-22347: address https://github.com/advisories/GHSA-qppj-fm5r-hxr3

Waiting for https://github.com/openshift/external-dns/pull/56 to be merged.

**Comment by melvinjoseph86** [comment]
Verified using clusterbot
melvinjoseph@mjoseph-mac Downloads % oc get clusterversion
NAME      VERSION                                                AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.12.0-0.test-2023-12-13-071635-ci-ln-8dm452k-latest   True        False         138m    Cluster version is 4.12.0-0.test-2023-12-13-071635-ci-ln-8dm452k-latest

melvinjoseph@mjoseph-mac Downloads % oc get csv
NAME                           DISPLAY                VERSION   REPLACES   PHASE
external-dns-operator.v1.0.0   ExternalDNS Operator   1.0.0                Succeeded
melvinjoseph@mjoseph-mac Downloads % oc get sub 
NAME       PACKAGE                 SOURCE     CHANNEL
oo-rppfr   external-dns-operator   oo-q5m9w   stable-v1.0
melvinjoseph@mjoseph-mac Downloads % oc get po
NAME                                     READY   STATUS    RESTARTS   AGE
external-dns-operator-84c4558f86-nvbpt   2/2     Running   0          55m
oo-q5m9w-kjffh                           1/1     Running   0          55m

melvinjoseph@mjoseph-mac Downloads % oc get po  external-dns-operator-84c4558f86-nvbpt -oyaml  
apiVersion: v1
items:
<----snip----->
spec:
  containers:
  - args:
    - --metrics-bind-address=127.0.0.1:8080
    - --operator-namespace=$(OPERATOR_NAMESPACE)
    - --operand-namespace=$(OPERATOR_NAMESPACE)
    - --externaldns-image=$(RELATED_IMAGE_EXTERNAL_DNS)
    - --trusted-ca-configmap=$(TRUSTED_CA_CONFIGMAP_NAME)
    - --leader-elect
    - --webhook-disable-http2
<-----snip----->
  - args:
    - --secure-listen-address=0.0.0.0:8443
    - --upstream=http://127.0.0.1:8080/
    - --logtostderr=true
    - --v=10
    - --tls-cert-file=/var/run/secrets/serving-cert/tls.crt
    - --tls-private-key-file=/var/run/secrets/serving-cert/tls.key
    - --http2-disable
<-----snip----->

**Comment by alebedev87** [comment]
/retitle [release-1.0] OCPBUGS-22347: address CVE-2023-44487

The operand image was bumped thanks to [the mirroring config](https://github.com/openshift/release/pull/46634).

File: bundle/manifests/external-dns-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -436,13 +436,14 @@ spec:
...
                 - --leader-elect
+                - --webhook-disable-http2
                 env:
                 - name: OPERATOR_NAMESPACE
                   valueFrom:
                     fieldRef:
                       fieldPath: metadata.namespace
                 - name: RELATED_IMAGE_EXTERNAL_DNS
-                  value: quay.io/openshift/origin-external-dns@sha256:419ae8594ba87f197c49e298ecec85e185ed29e8fd80e874351415c7ce69381e
+                  value: quay.io/external-dns-operator/external-dns@sha256:2ae9c9dc9a528443dc8d4edfca2b681a7376eff706f3b52feda763f6c4e25f07
```
Comment (gcs278) [inline bundle/manifests/external-dns-operator.clusterserviceversion.yaml:446]: So this is a direct result of https://github.com/openshift/release/pull/46634/files#diff-a99f743ad0309322b74e5b90fbb6779685260c4ea5d77c17fc5bce861a2f21f7?

Could you help me understand the context of this old image repo: `quay.io/openshift/origin-external-dns` . Is that some sort of default image? How does that differs from quay?

**Comment by gcs278** [comment]
No issue, just curiosity
/lgtm
/approve

**Comment by alebedev87** [comment]
/approve

works?

**Pull request merged**